File: Frontend/src/core/domain/models/Workflow.ts
  // File: src/core/domain/models/Workflow.ts
  export interface WorkflowStage {
    id: string;
    name: string;
    color: string;
    editable: boolean;
    visible: boolean;
  }
  
  export interface Workflow {
    id: string;
    user_id: string;
    user_email: string;
    stages: WorkflowStage[];
    stage_order: string[];
    default: boolean;
}

File: Backend/app/models/workflow.py
# app/models/workflow.py
from pydantic import BaseModel, Field, EmailStr
from typing import List, Optional
from bson import ObjectId

class WorkflowStage(BaseModel):
    id: str = Field(default_factory=lambda: str(ObjectId()))
    name: str
    color: str = "gray"
    editable: bool = True
    visible: bool = True
    
class Workflow(BaseModel):
    id: str = Field(default_factory=lambda: str(ObjectId()))
    user_id: Optional[str] = None  # Add user ID field
    user_email: Optional[EmailStr] = None  # Add user email field
    stages: List[WorkflowStage]
    stage_order: List[str]
    default: bool = False

    model_config = {
        "json_schema_extra": {
            "example": {
                "user_id": "user123",
                "user_email": "user@example.com",
                "stages": [
                    {"id": "unassigned", "name": "Unassigned", "color": "gray", "editable": False, "visible": True},
                    {"id": "resume-submitted", "name": "Resume Submitted", "color": "blue", "editable": True, "visible": True}
                ],
                "stage_order": ["unassigned", "resume-submitted"],
                "default": True
            }
        }
    }